core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckServiceBuilder.java

```diff
@@ -0,0 +1,234 @@
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.auth.HttpAuthService;
+
+/**
+ * Builds a {@link HealthCheckService}.
+ */
+public final class HealthCheckServiceBuilder {
+
+    private static final int DEFAULT_LONG_POLLING_TIMEOUT_SECONDS = 60;
+    private static final double DEFAULT_LONG_POLLING_TIMEOUT_JITTER_RATE = 0.2;
+
+    private final ImmutableSet.Builder<HealthChecker> healthCheckers = ImmutableSet.builder();
+    private AggregatedHttpResponse healthyResponse = AggregatedHttpResponse.of(HttpStatus.OK,
+                                                                               MediaType.JSON_UTF_8,
+                                                                               "{\"healthy\":true}");
+    private AggregatedHttpResponse unhealthyResponse = AggregatedHttpResponse.of(HttpStatus.SERVICE_UNAVAILABLE,
+                                                                                 MediaType.JSON_UTF_8,
+                                                                                 "{\"healthy\":false}");
+    private long maxLongPollingTimeoutMillis = TimeUnit.SECONDS.toMillis(DEFAULT_LONG_POLLING_TIMEOUT_SECONDS);
+    private double longPollingTimeoutJitterRate = DEFAULT_LONG_POLLING_TIMEOUT_JITTER_RATE;
+    @Nullable
+    private HealthCheckUpdateHandler updateHandler;
+
+    HealthCheckServiceBuilder() {}
+
+    /**
+     * Adds the specified {@link HealthChecker}s that determine the healthiness of the {@link Server}.
+     *
+     * @return {@code this}
+     */
+    public HealthCheckServiceBuilder checkers(HealthChecker... healthCheckers) {
+        return checkers(ImmutableSet.copyOf(requireNonNull(healthCheckers, "healthCheckers")));
+    }
+
+    /**
+     * Adds the specified {@link HealthChecker}s that determine the healthiness of the {@link Server}.
+     *
+     * @return {@code this}
+     */
+    public HealthCheckServiceBuilder checkers(Iterable<? extends HealthChecker> healthCheckers) {
+        this.healthCheckers.addAll(requireNonNull(healthCheckers, "healthCheckers"));
+        return this;
+    }
+
+    /**
+     * Sets the {@link AggregatedHttpResponse} to send when the {@link Service} is healthy. The following
+     * response is sent by default:
+     *
+     * <pre>{@code
+     * HTTP/1.1 200 OK
+     * Content-Type: application/json; charset=utf-8
+     *
+     * { "healthy": true }
+     * }</pre>
+     *
+     * @return {@code this}
+     */
+    public HealthCheckServiceBuilder healthyResponse(AggregatedHttpResponse healthyResponse) {
```

Comment:
Just a note, I don't think this will work if the content of the response is `ByteBufHttpData`, since it can only be read once. Since you shouldn't use that class without really knowing what's going on, I think it's ok but just want to make sure.

Reply:
Ah, yeah. I just made a copy.